fall-through-cache/fall-through-cache.js

```diff
@@ -72,9 +73,112 @@
+
 module.exports = connect.behavior("fall-through-cache",function(baseConnect){
+	var setExpando = function(map, prop, value) {
+		if("attr" in map) {
+			map[prop] = value;
+		} else {
+			map._data[prop] = value;
+		}
+	};
+	var getExpando = function(map, prop) {
+		if("attr" in map) {
+			return map[prop];
+		} else {
+			return map._data[prop];
+		}
+	};
+
+	var overwrite = function( connection, Constructor, prototype, statics) {
+		var prop;
+
+		for(prop in prototype) {
+			Constructor.prototype[prop] = prototype[prop](Constructor.prototype[prop], connection);
+		}
+		if(statics) {
+			for(prop in statics) {
+				Constructor[prop] = statics[prop](Constructor[prop], connection);
+			}
+		}
+	};
+
+	var addIsConsistent = function(connection, Constructor) {
+		overwrite(connection, Constructor, {
+
+			_setInconsistencyReason: function(base, connection) {
+				return function(error) {
+					var oldError = getExpando(this, "inconsistencyReason");
+
+					setExpando(this, "inconsistencyReason", error);
+					canEvent.dispatch.call(this, "inconsistencyReason", [error, oldError]);	
+				}
+			},
+
+			_setIsConsistent: function(base, connection) {
+				return function(isConsistent) {
+					setExpando(this, "_isConsistent", isConsistent);
+					canEvent.dispatch.call(this, "_isConsistent", [isConsistent, !isConsistent]);
+				};
+			},
+      /**
+       * @function can-connect/fall-through-cache/fall-through-cache.isConsistent isConsistent
+       * @parent can-connect/fall-through-cache/fall-through-cache.mixins
+       *
+       * Returns whether or not the data is consistent between the server and 
+       * the fall-through-cache data.
+       *
+       * @signature `[map|list].isConsistent()`
+       *
+       *   Returns true if the data has been successfully returned from the 
+       *   server and in sync with the fall-through-cache. Returns false if the
+       *   data is from the cache.
+       *   
+       *   @return {Boolean}
+       */
+			isConsistent: function(base, connection) {
+				return function() {
+					Observation.add(this,"_isConsistent");
+		    	return !!getExpando(this, "_isConsistent");
+				};
+			}
+      /**
+       * @function can-connect/fall-through-cache/fall-through-cache.inconsistencyReason inconsistencyReason
+       * @parent can-connect/fall-through-cache/fall-through-cache.mixins
+       *
+       * Returns the error of the AJAX call from the base data layer.
+       *
+       * @signature `[map|list].inconsistencyReason`
+       *
+       *   Returns the error fo the base data layer's AJAX call if it's promise
+       *   was rejected. If there isn't an inconsistency issue between the server
+       *   and fall-through-cache layer, this will be undefined.
+       *   
+       *   @return {Object}
+       */
+		});
+	};
 
 	var behavior = {
+		init: function() {
+      // If List and Map are on the behavior, then we go ahead and add the 
+      // isConsistent API information.
```

Comment:
Off the top of my head no. Based on the suggested requirements, these features would be specific to maps and lists, and made sense to me that if we wanted to have these function features available in templates.

Do you have any suggestions?